Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs

```diff
@@ -142,53 +142,71 @@ private void PopUpContextMenu ()
 				});
 			}
 
+			FocusClickedRow ();
+
 			var menuOrigin = this.Superview.ConvertPointToView (new CGPoint (Frame.Location.X - 1, Frame.Location.Y + Frame.Size.Height + 4), null);
 
 			var popupMenuEvent = NSEvent.MouseEvent (NSEventType.LeftMouseDown, menuOrigin, (NSEventModifierMask)0x100, 0, this.Window.WindowNumber, this.Window.GraphicsContext, 0, 1, 1);
 
 			NSMenu.PopUpContextMenu (popUpContextMenu, popupMenuEvent, this);
 		}
 
+		private void FocusClickedRow ()
+		{
+			if (Superview != null
+				&& (Superview.Subviews.Length > 1)) {
+				NSView view = Superview.Subviews[1];
```

Comment:
This seems very fragile, why don't we search the subviews for a `PropertyEditorControl` instead?

Reply:
We can.

Reply:
Found an issue where it wasn't focusing all the rows. A more recursive solution seems to work.